grails-app/controllers/org/pih/warehouse/JsonController.groovy

```diff
@@ -1846,12 +1849,15 @@ class JsonController {
                             destination             : it.destination,
                             productCode             : it.productCode,
                             productName             : it.productName,
-                            unitPrice               : it.unitPrice ?: '',
+                            productFamily           : it.productFamily,
+                            category                : it.category,
+                            productCatalogs         : it.productCatalogs,
+                            unitPrice               : it.unitPrice,
                             qtyRequested            : it.quantityRequested,
                             qtyIssued               : it.quantityIssued,
                             qtyDemand               : it.quantityDemand,
-                            reasonCode              : it.reasonCode ?: '',
```

Comment:
Why were "Elvis operators" removed in this file?

Reply:
I think Alan followed my suggestion about `productFamily` and moved them (elvis operators) to the service layer for every property. 
Keeping it in both  - `JsonController` and `ForecastingService` you just do twice the same, so you might run into controller having: `'' ?: ''`